sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -873,54 +891,9 @@ export class MessageReceiver extends LinkEntity {
       }
 
       // We should attempt to reopen only when the receiver(sdk) did not initiate the close
-      let shouldReopen = false;
-      if (receiverError && !wasCloseInitiated) {
-        const translatedError = translate(receiverError);
-        if (translatedError.retryable) {
-          shouldReopen = true;
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is retryable. This is a candidate for re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        } else {
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is NOT retryable. Hence NOT re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        }
-      } else if (!wasCloseInitiated) {
-        shouldReopen = true;
-        log.error(
-          "[%s] close() method of Receiver '%s' with address '%s' was not called. " +
-            "There was no accompanying error as well. This is a candidate for re-establishing " +
-            "the receiver link.",
-          connectionId,
-          this.name,
-          this.address
-        );
-      } else {
-        const state: any = {
-          wasCloseInitiated: wasCloseInitiated,
-          receiverError: receiverError,
-          _receiver: this._receiver
-        };
-        log.error(
-          "[%s] Something went wrong. State of Receiver '%s' with address '%s' is: %O",
-          connectionId,
-          this.name,
-          this.address,
-          state
-        );
-      }
-      if (shouldReopen) {
+      const shouldReopen = !receiverError || translate(receiverError).retryable;
+
+      if (shouldReopen && !this.wasCloseInitiated) {
```

Comment:
Why not move the `!this.wasCloseInitiated` to the previous line when trying to decide what `shouldReopen` be?

Reply:
Updated to exit early if `this.wasCloseInitiated` is true.